src/compiler/parser.ts

```diff
@@ -3337,7 +3337,10 @@ namespace ts {
         function parseAwaitExpression() {
             const node = <AwaitExpression>createNode(SyntaxKind.AwaitExpression);
             nextToken();
-            node.expression = parseSimpleUnaryExpression();
+            node.expression = token === SyntaxKind.AwaitKeyword
+                // Forbid `await await`

```

Comment:
> Currently it is not possible for an *AwaitExpression* and a *YieldExpression* to be parsed together, so technically passing the ?Yield parameter for *UnaryExpression*: <sub>[+Await]</sub>`await` *UnaryExpression*<sub>[?Yield, Await]</sub> is unnecessary. It will probably be useful for AsyncGenerators however.

`yield` expressions and `await` expressions can't be mixed because since async generators aren't really a thing at this point, so whether or not it's a valid parse is kind of moot.

**Keep in mind**, `await yield` is valid in a plain `async` function. `yield` will be parsed as an identifier right now.

If you *did* have asynchronous generators, `yield await 10` would be valid in those contexts, but `await yield` would not - you'd need to parenthesized the `yield` keyword.